llvm/lib/Target/AMDGPU/SIPeepholeSDWA.cpp

```diff
@@ -1361,6 +1379,499 @@ bool SIPeepholeSDWALegacy::runOnMachineFunction(MachineFunction &MF) {
   return SIPeepholeSDWA().run(MF);
 }
 
+static bool isSrcDestFP16Bits(MachineInstr *MI, const SIInstrInfo *TII) {
```

Comment:
Is there a way to tablegen this switch-case?

(Also, are there relevant things that use the OPSEL scheme and not the SDWA one?)

Reply:
This switch overlaps a lot with zeroesHigh16BitsOfDest, but I also suspect you don't actually need this information